internal/mesh/internal/types/proxy_state_template.go

```diff
@@ -61,3 +66,138 @@ func RegisterProxyStateTemplate(r resource.Registry) {
 		},
 	})
 }
+
+func ValidateProxyStateTemplate(res *pbresource.Resource) error {
```

Comment:
* Do we need to wire this up into the register call above?
* Could you add some tests for this?

Reply:
:facepalm:

Reply:
done and done